packages/smooth_app/lib/pages/product/picture_capturer.dart

```diff
@@ -0,0 +1,165 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:image_cropper/image_cropper.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/database/product_query.dart';
+import 'package:smooth_app/generic_lib/buttons/smooth_action_button.dart';
+import 'package:smooth_app/generic_lib/dialogs/smooth_alert_dialog.dart';
+import 'package:smooth_app/helpers/ui_helpers.dart';
+
+class PictureCapturer extends StatefulWidget {
+  const PictureCapturer({required this.barcode, required this.imageType});
+
+  final ImageField imageType;
+  final String barcode;
+
+  @override
+  State<PictureCapturer> createState() => _PictureCapturerState();
+}
+
+class _PictureCapturerState extends State<PictureCapturer> {
+  late File? picture;
+  late Future<void> photoPicker;
+
+  @override
+  void initState() {
+    super.initState();
+    photoPicker = _getImage();
+  }
+
+  Future<void> _getImage() async {
```

Comment:
Not sure it's the most elegant way to do that, a page with a `Future`.

From whoever calls `PictureCapturer`, couldn't we call `_getImage` and then only if `picture != null` call `PictureCapturer` with no `Future`?

That would mean a more readable helper method `Future<File> pickImageAndCrop`. And a more readable page `PictureCapturer` with no `Future`, to be called when the result of `pickImageAndCrop` is not null.

Does that make sense?

Reply:
Ummm. I think this is generally a good advice, and something I already thought about and I agree doing away with the FutureBuilder stuff will indeed be nice.

The only thing is that there's a `Retake` button on the PictureCapturer screen, when pressed it should restart the `pickAndCrop` method, that's why I wanted it to be part of this class. I guess I could still do what you are suggesting, but in a convoluted way, let me give it a shot and see what comes out of it.

Reply:
Done. PTAL